pensieve/pensieve.py

```diff
@@ -146,10 +146,22 @@ def paragraphs(self):
         if not self._paragraphs:
             print('Generating paragraphs for doc '+str(self.id))
             self._paragraphs = []
-            for i, p in enumerate(self.text.split('\n')):
-                if len(p.split(' ')) < 30:
+            list = self.text.split('\n')
+            i = 0
+            while i < len(list):
+                if len(list[i].strip()) == 0:
+                    i += 1
                     continue
-                self._paragraphs.append(Paragraph(p, i, self))
+                if len(list[i].split(' ')) >= 25:
+                    self._paragraphs.append(Paragraph(list[i], i, self))
+                else:
+                    chunk = ""
+                    while (i < len(list)) and (len(list[i].split(' ')) < 25 or list[i][0] == "'" or list[i][0] == '"'):
```

Comment:
Using a for loop + `enumerate` should simplify this logic too (i.e. you won't need to check where you are in the loop)